src/imap.rs

```diff
@@ -450,12 +469,11 @@ impl Imap {
                             info!(context, "close/expunge succeeded");
                         }
                         Err(err) => {
-                            warn!(context, "failed to close session: {:?}", err);
-                            return ImapActionResult::Failed;
+                            return Err(SelectError::CloseExpungeFailed(err.to_string()));
```

Comment:
Put `err` instead of `err.to_string()` there.

Reply:
wouldn't the caller then need to deal with error coming from async_imap? I'd think that's a bit much here.

Reply:
It will just ignore the underlying cause like it ignores the `String` now.

Reply:
done.